submissions/svart0222/port-sniffer/index.js

```diff
@@ -0,0 +1,75 @@
+const net = require('net');
+
+const checkPort = (port, host) => {
+  return new Promise(resolve => {
+    const socket = net.createConnection(port, host, () => {
+      clearTimeout(timer);
+      resolve({ port, result: true });
+      socket.end();
+    });
+    const timer = setTimeout(() => {
+      resolve({ port, result: false });
+      socket.end();
+    }, 300);
+    socket.on('error', () => {
+      clearTimeout(timer);
+      resolve({ port, result: false });
+    });
+  });
+};
+
+const getResult = async (host, startPoint = 0, endPoint = 65535) => {
+  try {
+    const result = [];
+    let currentPoint = startPoint;
+
+    while (currentPoint < endPoint) {
```

Comment:
Try to replace while loop, on for loop, and use break and continue statements. 
While it is a blocking operation, and not required in this case. 

Also, checkPort function returns a promise, that should be resolved, you need to resolve it somehow, and push to results only actual results

Reply:
Sorry, can't understand why for loop is better than while loop ? 
if I correctly understand you recommend wait result from every check, yes ?